compileoptions.cmake

```diff
@@ -66,6 +66,8 @@ if(CLR_CMAKE_PLATFORM_UNIX_ARM)
      add_definitions(-DARM_SOFTFP)
      add_compile_options(-mfloat-abi=softfp)
      add_compile_options(-target armv7-linux-gnueabi)
+   elseif(CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+     add_compile_options("-march=armv7-a")
    else()
      add_compile_options(-target armv7-linux-gnueabihf)
```

Comment:
Is it intentional that `-target armv7-linux-gnueabihf` is not longer going to be present for Alpine?

Reply:
I also wonder how come you don't need to specify the target.

Reply:
Yes it is intentional. Alpine has `armv6-alpine-linux-musleabi` no GNU stuff, which that sets arch to armv6 which CoreCLR does not support, so we only need to set march to armv7 and abi targets gets picked up on ARM32 system.

Reply:
Would similar thing work on other platforms - that means, if we used just the `-march` instead of the `-target`?

Reply:
I think the difference in case of Alpine+CoreCLR is that Alpine provides v6 ABI target for ARM32 that has arch set to armv6 by default that works on both ARMv6 and v7 versions of OS, while other platforms have specific armv7 ABI with default arch set correctly to armv7. We can test if setting `-march=armv7-a` is enough for those.

As noted in the issue that I was actually stuck to get pass 1% of build, asked a question https://github.com/richfelker/musl-cross-make/issues/43 and Rich Felker (primary author of musl libc https://ewontfix.com/about/) told me how to deal with this using -march. It turned out to be working well (until 26% in 4 hours of compilation over qemu).

At 26%, I get:

```
[ 25%] Building CXX object src/debug/createdump/CMakeFiles/createdump_lib.dir/createdump.cpp.o
/coreclr/src/gc/unix/cgroup.cpp:407:32: error: implicit conversion from 'unsigned long long' to 'size_t'
      (aka 'unsigned int') changes value from 18446744073709551615 to 4294967295 [-Werror,-Wconstant-conversion]
    size_t rlimit_soft_limit = RLIM_INFINITY;
           ~~~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~
/usr/include/sys/resource.h:72:24: note: expanded from macro 'RLIM_INFINITY'
#define RLIM_INFINITY (~0ULL)
                       ^~~~~
1 error generated.
make[2]: *** [src/gc/CMakeFiles/clrgc.dir/build.make:495: src/gc/CMakeFiles/clrgc.dir/unix/cgroup.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:61853: src/gc/CMakeFiles/clrgc.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 26%] Building CXX object src/debug/createdump/CMakeFiles/createdump_lib.dir/crashinfo.cpp.o
[ 26%] Building CXX object src/debug/createdump/CMakeFiles/createdump_lib.dir/threadinfo.cpp.o
[ 26%] Building CXX object src/debug/createdump/CMakeFiles/createdump_lib.dir/datatarget.cpp.o
/coreclr/src/debug/createdump/threadinfo.cpp:6:10: fatal error: 'asm/ptrace.h' file not found
#include <asm/ptrace.h>
         ^~~~~~~~~~~~~~
1 error generated.
make[2]: *** [src/debug/createdump/CMakeFiles/createdump_lib.dir/build.make:111: src/debug/createdump/CMakeFiles/createdump_lib.dir/threadinfo.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:61988: src/debug/createdump/CMakeFiles/createdump_lib.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
Failed to build CoreCLR component.
```

I am planning to work on those errors separately after this one gets through.